pkg/xds/generator/secrets_generator.go

```diff
@@ -38,5 +43,39 @@ func (s SecretsProxyGenerator) Generate(ctx xds_context.Context, proxy *core_xds
 			Resource: caSecret,
 		},
 	)
+
+	return nil
+}
+
+func (s SecretsProxyGenerator) Generate(
+	ctx xds_context.Context,
+	proxy *core_xds.Proxy,
+) (*core_xds.ResourceSet, error) {
+	resources := core_xds.NewResourceSet()
+
+	if proxy.GetMeshes != nil {
+		for _, mesh := range proxy.GetMeshes().Items {
+			if !mesh.MTLSEnabled() {
+				continue
+			}
+
+			if err := createSecrets(resources, ctx.ControlPlane.Secrets, proxy, mesh); err != nil {
+				return nil, err
+			}
+		}
+
+		return resources, nil
+	}
+
+	mesh := ctx.Mesh.Resource
+
+	if !mesh.MTLSEnabled() {
+		return nil, nil
+	}
+
+	if err := createSecrets(resources, ctx.ControlPlane.Secrets, proxy, mesh); err != nil {
+		return nil, err
+	}
+
```

Comment:
Can we delete this code and use only meshes from `GetMeshes`? So if you generate a configuration for DPP then `GetMeshes` returns 1 mesh. When you generate a configuration for Egress then `GetMeshes` returns all meshes.

Reply:
@bartsmykla Could you help me to understand why `GetMeshes` is a function for a list of meshes?

I think `GetMeshes` should be converted to just a list like `SupportedMeshes []*core_mesh.MeshResource`. `GetMeshes` as a function sounds like there will be an alternative implementation that will load meshes from somewhere, but right now we are trying to change xds generators to be purely CPU-bound without touching I/O. 

The second thing that I'd like to ask for is this
```
	// todo(lobkovilya): split Proxy struct into DataplaneProxy and IngressProxy
	// TrafficRouteList is used only for generating configs for Ingress.
	TrafficRouteList *core_mesh.TrafficRouteResourceList
```
When creating `ZoneEgress`, let's try to create `EgressProxy` with relevant fields to Egress.

Reply:
After a discussion with Jakub I confirmed we can safely put a slice of meshes instead of a function and decided to create a separate struct for `ZoneEgressProxy`, where in the future we would put `ZoneEgress` resource (when created), in the future we could refactor other parts of this struct in a way, where anything Ingress related would be placed under `ZoneIngressProxy` and anything Dataplane related would go under `DataplaneProxy`, like:

```go
type Proxy struct {
	Id         ProxyId
	APIVersion envoy_common.APIVersion

	DataplaneProxy   *DataplaneProxy
	ZoneIngressProxy *ZoneIngressProxy
	ZoneEgressProxy  *ZoneEgressProxy
}
```